src/plugins/intel_gpu/src/runtime/memory_pool.cpp

```diff
@@ -309,25 +319,146 @@ void memory_pool::clear_pool_for_network(uint32_t network_id) {
+            << "MB, type: " << mem.second._type << ")'s users: " << std::endl;
+        total_mem_pool_size += mem.first;
+        float min_percentage = 100.0f;
+        float max_percentage = 0.f;
         for (auto user : mem.second._users) {
-            GPU_DEBUG_COUT << "   --- " << user._prim_id << " (" << user._unique_id << ") " << std::endl;
+            float percentage = get_percentage(user._mem_size, mem.first);
+            min_percentage = std::min(percentage, min_percentage);
+            max_percentage = std::max(percentage, max_percentage);
+            total_mem_request += static_cast<float>(user._mem_size);
+            GPU_DEBUG_COUT << "    --- " << user._prim_id << " (" << user._unique_id << "), "
+                << get_mb_size(user._mem_size) << "MB, " << percentage << "%" << std::endl;
         }
+        GPU_DEBUG_COUT <<  "   - min of the memory pool entry: " << min_percentage << std::endl;
+        GPU_DEBUG_COUT <<  "   - max of the memory pool entry: " << max_percentage << std::endl;
+    }
+    if (total_mem_pool_size > 0.f) {
+        GPU_DEBUG_COUT << "========== non-padded pool efficiency: "
+            << std::to_string(static_cast<float>(total_mem_request / total_mem_pool_size))
+            << " (Total memory request : " << get_mb_size(total_mem_request) << "MB"
+            << " / Total memory pool size : " << get_mb_size(total_mem_pool_size) << "MB) ==========" << std::endl;
     }
+    float total_non_padded_pool_size = total_mem_pool_size;
     GPU_DEBUG_COUT << "========== padded pool (" << _padded_pool.size() << " records) ==========" << std::endl;
+    total_mem_request = 0.f;
+    total_mem_pool_size = 0.f;
     for (auto mem : _padded_pool) {
-        GPU_DEBUG_COUT << " layout: " << mem.first.to_short_string() << std::endl;
+        GPU_DEBUG_COUT << " layout: " << mem.first.to_short_string() << ", records(" << mem.second.size() << ")" << std::endl;
         for (auto record : mem.second) {
-            GPU_DEBUG_COUT << "    " << record._memory->buffer_ptr() << ", type: " << record._type << ", users : " << std::endl;
+            size_t mem_size = record._memory->size();
+            GPU_DEBUG_COUT << "  " << record._memory->buffer_ptr() << " (size:" << get_mb_size(mem_size)
+                            << "MB, type: " << record._type << ")'s usrs : " << std::endl;
+            float min_percentage = 100.0f;
+            float max_percentage = 0.f;
+            total_mem_pool_size += mem_size;
             for (auto user : record._users) {
-                GPU_DEBUG_COUT << "    --- " << user._prim_id << " (" << user._unique_id << std::endl;
+                float percentage = get_percentage(user._mem_size, mem_size);
+                min_percentage = std::min(percentage, min_percentage);
+                max_percentage = std::max(percentage, max_percentage);
+                total_mem_request += static_cast<float>(user._mem_size);
```

Comment:
Seems that total mem request is calculated many times, I'dl like to recommend it to be stored.

Reply:
Done.